testing_and_setup/compass/manage_regression_suite.py

```diff
@@ -193,19 +196,282 @@ def process_test_clean(test_tag, work_dir):  # {{{
+    local_parallel_code += "  Queue_running = []\n"
+    local_parallel_code += "  index = 0\n"
+    local_parallel_code += "  continue_add = True\n"
+    local_parallel_code += "  Done = False\n"
+    local_parallel_code += "  while True:\n"
+    local_parallel_code += "     if continue_add and not Done:\n"
+    local_parallel_code += "      if number_of_procs >= procs[index]:\n"
+    local_parallel_code += "        try:\n"
+    local_parallel_code += "          if index < len(command):\n"
+    local_parallel_code += "#            print('-----IN ADDING PHASE-----')\n"
+    local_parallel_code += "#            print('we have {} number_of_procs, current task uses {} Adding to queue'.format(number_of_procs, procs[index]))\n"
+    local_parallel_code += "            case_output = open('case_outputs/'+command[index][0].replace('/', '_'), 'w+')\n"
+    local_parallel_code += "#            print('processing command: {}'.format(command[index][1]))\n"
+    local_parallel_code += "            os.chdir(command[index][0])\n"
+    local_parallel_code += "            command[index][1][2] = command[index][1][2] +'/'+data_dic['run_file_names'][index]\n"
+    local_parallel_code += "            open_proc = subprocess.Popen(command[index][1], stdout=case_output, stderr=case_output)\n"
+    local_parallel_code += "            os.chdir(base)\n"
+    local_parallel_code += "            Queue_running.append([open_proc, procs[index],command[index][1]]) \n"
+    local_parallel_code += "            number_of_procs = number_of_procs - procs[index]\n"
+    local_parallel_code += "#            print('       New number_of_procs: {}'.format(number_of_procs)) \n"
+    local_parallel_code += "        except subprocess.CalledProcessError: \n"
+    local_parallel_code += "          print('error for : {}'.format(str(command[index][1]))) \n"
+    local_parallel_code += "      elif number_of_procs < procs[index]: \n"
+    local_parallel_code += "#        print('NOT ENOUGH WAIT FOR PROCESSES') \n"
+    local_parallel_code += "        continue_add = False \n"
+    local_parallel_code += "      index = index + 1 \n"
+    local_parallel_code += "      if index > len(procs) -1: \n"
+    local_parallel_code += "#        print('No more to add moving to processing phase') \n"
+    local_parallel_code += "        continue_add = False \n"
+    local_parallel_code += "     elif not continue_add and not Done: \n"
+    local_parallel_code += "       for background_process in Queue_running: \n"
+    local_parallel_code += "         background_process[0].wait() \n"
+    local_parallel_code += "         pid =background_process[0].pid \n"
+    local_parallel_code += "         if not psutil.pid_exists(pid): \n"
+    local_parallel_code += "           Queue_running.remove(background_process) \n"
+    local_parallel_code += "           name = background_process[-1][-1]\n"
+    local_parallel_code += "           name = name[name.find('compass')+len('compass')+1:-3].replace('/','_')\n" 
```

Comment:
I don't think this is a safe way to determine the name of the task that was running.  If you use a `work_dir` there will be no `compass` directory (at least not typically).  I think we can figure this out some other way.  For example, we could store the pid in the dictionary for a given  test case within the `testcases` data structure.